### PR TITLE
fix remove duplicated method

### PR DESCRIPTION
**Pull request by bakridemos**
Description: 

- move the method 'generateDownloadFilename' to a new service 'NameGenerator.php'
- remove duplicated method

Note: these changes are not made randomly, this method can be used by an addon in the future, otherwise it was duplicated and it will be better to move it to a service

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Controller/Forum/DemosPlanReleaseController.php [hunk 1]
```diff
@@ -841,7 +845,7 @@ public function exportReleaseAction(Environment $twig, string $releaseId)
         $response->setContent($bom.$response->getContent());
         $filename = 'export_'.$part.'_'.date('Y_m_d_His').'.csv';
         $response->headers->set('Content-Type', 'text/csv');
-        $response->headers->set('Content-Disposition', $this->generateDownloadFilename($filename));
+        $response->headers->set('Content-Disposition', $pdfNameService->generateDownloadFilename($filename));
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Controller/Forum/DemosPlanReleaseController.php:848]: Why did you choose PdfNameService? In this case it is used to generate a name for a csv file which seems weird

Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Controller/Forum/DemosPlanReleaseController.php]: I don't think so as in this case the service name would be weird when generating Filenames for pdf files. I think it needs to be something neutral

Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Controller/Forum/DemosPlanReleaseController.php]: done in https://github.com/demos-europe/demosplan-core/pull/1573/commits/ec3512bd9a27ab7cfbcb347b1555ce7ef448cced

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 2]
```diff
@@ -1669,8 +1672,12 @@ protected function replacePhaseByPhaseNameForVotedStatementList(array $statement
      *
      * @throws Exception
      */
-    protected function createPdfDraftStatement($draftStatementList, $type, Procedure $procedure, $itemsToExport = null)
-    {
+    protected function createPdfDraftStatement(
+        $draftStatementList, $type,
+        Procedure $procedure,
+        $itemsToExport = null,
+        NameGenerator $nameGenerator
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php]: fixed https://github.com/demos-europe/demosplan-core/pull/1573/commits/150e36db2db282d7a4da63f9a3a47aeadc87ff98

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 3]
```diff
@@ -1669,8 +1672,12 @@ protected function replacePhaseByPhaseNameForVotedStatementList(array $statement
      *
      * @throws Exception
      */
-    protected function createPdfDraftStatement($draftStatementList, $type, Procedure $procedure, $itemsToExport = null)
-    {
+    protected function createPdfDraftStatement(
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:1675]: You need to modify the method calls as well when you change the signature

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php (hunk 3 above)
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:1685]: sorry fixed here https://github.com/demos-europe/demosplan-core/pull/1573/commits/058f6f64501dd13389225679e170f073ed202a6e

**Review by mussbach — CHANGES_REQUESTED** [review]
As this introduces a shiny, tiny and easy testable class you should add a test

**Comment by bakridemos** [comment]
> As this introduces a shiny, tiny and easy testable class you should add a test

done in https://github.com/demos-europe/demosplan-core/pull/1573/commits/05f1648cb481f70abe5d19928a66c415a94042bc

**Pull request merged**